Project2015To2017/Reading/ProjectReader.cs

```diff
@@ -1,28 +1,48 @@
+using Project2015To2017.Definition;
 using System;
 using System.Collections.Generic;
 using System.Collections.Immutable;
 using System.IO;
 using System.Linq;
 using System.Xml;
 using System.Xml.Linq;
-using Project2015To2017.Definition;
 using static Project2015To2017.Definition.Project;
 
 namespace Project2015To2017.Reading
 {
 	public class ProjectReader
 	{
-		public Project Read(string filePath)
+		public static readonly ProjectReader Instance = new ProjectReader();
+		public bool EnableCaching { get; set; }
+
+		public Project Read(string filePath, IProgress<string> progress = null)
 		{
-			return Read(filePath, new Progress<string>(_ => { }));
+			progress = progress ?? new Progress<string>(_ => { });
+			return Read(new FileInfo(filePath), progress);
 		}
 
-		public Project Read(string filePath, IProgress<string> progress)
+		/// <summary>
+		/// Process-lifetime-long cache of loaded projects
+		/// <remarks>
+		/// It was supposed to be of <see cref="System.Runtime.Caching.ObjectCache"/> type, but that can still remove items from cache under memory pressure,
+		/// while we still need availability guarantees.
+		/// </remarks>
+		/// </summary>
+		private readonly Dictionary<int, Project> _cache = new Dictionary<int, Project>();
+
+		public Project Read(FileInfo file, IProgress<string> progress, bool loadModern = false)
```

Comment:
3rd parameter for future PR (yeah, that's support for modern CPS project reading).

Reply:
I like where I think that's going. I've converted a few projects and then wished I could run improved versions of my custom transformations, or take advantage of improvements to the converter.